### PR TITLE
Generalize impl EncodeAsVarULE for Cow

### PR DESCRIPTION
**Pull request by CAD97**
Missed in #2376

r? @sffc https://github.com/unicode-org/icu4x/pull/2376#pullrequestreview-1072276595

### CODE REVIEW

**Review by Manishearth — APPROVED** [review]

File: utils/zerovec/src/ule/encode.rs [hunk 1]
```diff
@@ -113,7 +113,7 @@ unsafe impl<T: VarULE + ?Sized> EncodeAsVarULE<T> for &'_ T {
 
 unsafe impl<T: VarULE + ?Sized> EncodeAsVarULE<T> for Cow<'_, T>
 where
-    T: Clone,
+    T: ToOwned,
```
Comment (sffc) [inline utils/zerovec/src/ule/encode.rs:116]: Please also make it `+ ?Sized` so that `Cow<str>` works

**Review by sffc — CHANGES_REQUESTED** [review]

File: utils/zerovec/src/ule/encode.rs (hunk 1 above)
Comment (CAD97) [reply utils/zerovec/src/ule/encode.rs:116]: It's already `<T: VarULE + ?Sized>`

Comment (sffc) [reply utils/zerovec/src/ule/encode.rs:116]: ok! It's a bit confusing to have bounds in two different places but you are correct.

**Review by sffc — APPROVED** [review]

File: utils/zerovec/src/ule/encode.rs (hunk 1 above)
Comment (CAD97) [reply utils/zerovec/src/ule/encode.rs:116]: There actually is a reason I did it like this: `<T: VarULE + ?Sized>` matches all of the other impls, and the `where` is the extra bounds. I didn't jump around for more examples to try to better identify what'd match existing style better.

(In my own code I tend to prefer putting `?Sized` inline and everything else in `where` bounds once things go beyond a single trait bound for a single generic type.)

**Pull request merged**